pwiz_tools/Skyline/SkylineFiles.cs

```diff
@@ -321,11 +331,41 @@ public bool OpenFile(string path, FormEx parentWindow = null)
             }
             catch (Exception x)
             {
-                exception = x;
-                // Was that even a Skyline file?
-                if (!SrmDocument.IsSkylineFile(path, out var explained))
+                var ex = x;
+                var isAuditLogException = false;
+                var msgStrings = new List<string>();
+                
+                do
                 {
-                    exception = new IOException(explained); // Offer a more helpful explanation than that from the failed XML parser
+                    //Make the error dialog a bit more user friendly by showing the full chain of nested exceptions in the message
+                    //so they don't have to dig through the stack traces to find the root cause.
+                    if(msgStrings.Count > 0)
+                        msgStrings.Add(Resources.ExceptionDialog_Caused_by_____);
+                    msgStrings.Add(ex.Message);
+                    if (ex.GetType() == typeof(AuditLogException))
+                    {
+                        isAuditLogException = true;
+                    }
+                    
+                    ex = ex.InnerException;
+
+                } while (ex != null);
+
+                if (isAuditLogException)
+                {
+                    MessageDlg.ShowWithException(parentWindow ?? this, 
+                        TextUtil.LineSeparate(msgStrings),

```

Comment:
Probably move the logic to a static function on AudiLogException, since it seems like it only really has an impact when AuditLogException is involved. Maybe 2 functions AuditLogException.IsAuditLogInvolved(Exception e) and AuditLogException.GetMultiLevelMessage(Exception e).